client/pages/park-list.jsx

```diff
@@ -27,28 +27,50 @@ export default class ParkList extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      parks: []
+      parks: [],
+      spinner: 'spinner-border medium-blue',
+      display: ''
     };
   }
 
   componentDidMount() {
-    fetch('/api/parks')
+    const { stateCode, path } = this.props;
+    let url;
+    if (path === 'all-parks') {
+      url = '/api/parks';
+    } else if (path === 'parks-by-state') {
+      url = `/api/parks/${stateCode}`;
+    }
+    fetch(url)
       .then(response => response.json())
       .then(data => {
         this.setState({
-          parks: data.data
+          parks: data.data,
+          spinner: 'spinner-border medium-blue d-none'
         });
       })
       .catch(error => {
         console.error('Error:', error);
       });
+    if (stateCode) {
+      this.setState({
+        display: `Parks in ${stateCode.toUpperCase()}`
+      });
+    } else {
+      this.setState({
+        display: 'All Parks'
```

Comment:
This does not really need to be in state, you can decide what do show based on `this.props.stateCode` in your `render()` method.